src/ucp/api/ucp.h

```diff
@@ -721,10 +737,11 @@ ucs_status_t ucp_worker_get_efd(ucp_worker_h worker, int *fd);
  *   with @ref UCP_FEATURE_WAKEUP to select proper transport
  *
  * @param [in]  worker    Worker to wait for events on.
+ * @param [in]  adderess  If non-null - will also wait for modifications to the address.
  *
  * @return Error code as defined by @ref ucs_status_t
  */
-ucs_status_t ucp_worker_wait(ucp_worker_h worker);
+ucs_status_t ucp_worker_wait(ucp_worker_h worker, void *address);
```

Comment:
Don't you think we should have an extensible API for monitoring more than one address, possibly with a limit stated as a capability? I know that there are plans to support more than one address at a time. Also, if FDs are an alternative to internal wait - the API should reflect this and allow for the same events to be monitored on both (or at least allow vendors to implement such support). I'll ask around to see if others want this.